ssl/t1_lib.c

```diff
@@ -2451,7 +2436,7 @@ static int ssl_scan_serverhello_tlsext(SSL *s, PACKET *pkt, int *al)
         else if (type == TLSEXT_TYPE_application_layer_protocol_negotiation) {
             unsigned len;
             /* We must have requested it. */
-            if (s->alpn_client_proto_list == NULL) {
+            if (!s->s3->alpn_sent || s->alpn_client_proto_list == NULL) {
```

Comment:
Isn't it sufficient to check `s->s3->alpn_sent` and omit the check for the proto list?